src/modules/follow_target_estimator/TargetEstimator.cpp

```diff
@@ -0,0 +1,317 @@
+		// Reset XY-acceleration while GPS is stale to avoid filter runoffs
+		_filter_states.a_ned_est = matrix::Vector3f{0.0f, 0.0f, 0.0f};
+	}
+
+	if (vel_fusion_old_enough && ((target_moving && !gps_vel_stale) || !target_moving)) {
+		// Wait with first velocity fusion until at least one position fusion has been done
+		if (PX4_ISFINITE(_filter_states.x_ned_est(0)) && PX4_ISFINITE(_filter_states.x_ned_est(1))
+		    && PX4_ISFINITE(_filter_states.x_ned_est(2))) {
+
+			const float dt_update_vel = (follow_target.timestamp - _last_velocity_fusion_timestamp) / 1000000.0f; // seconds
+			_last_velocity_fusion_timestamp = follow_target.timestamp;
+
+			velocity_meas_update(dt_update_vel, v_measured, &filter_states_prev, &_filter_states);
+			_v_measurement_old = v_measured;
+		}
+	}
+}
+
+void TargetEstimator::position_meas_update(float deltatime, matrix::Vector3f x_measured,
+		filter_states_s const *const filter_states_prev, filter_states_s *const filter_states) const
+{
+	// Position update
+	filter_states->x_ned_est = filter_states_prev->x_ned_est + _filter_gains.G_p * (x_measured -
+				   filter_states_prev->x_ned_est);
+
+
+	// Velocity update
+	filter_states->v_ned_est = filter_states_prev->v_ned_est + _filter_gains.H_p / (deltatime) *
+				   (x_measured - filter_states_prev->x_ned_est);
+
+	// Acceleration update
+	filter_states->a_ned_est = filter_states_prev->a_ned_est + 2.0f * _filter_gains.K_p / (deltatime * deltatime) *
+				   (x_measured - filter_states_prev->x_ned_est);
+	filter_states->saturate_acceleration(ACCELERATION_SATURATION);
+}
+
+void TargetEstimator::velocity_meas_update(float deltatime, matrix::Vector3f v_measured,
+		filter_states_s const *const filter_states_prev,
+		filter_states_s *const filter_states) const
+{
+	// Velocity update
+	filter_states->v_ned_est = filter_states_prev->v_ned_est + _filter_gains.G_v * (v_measured -
+				   filter_states_prev->v_ned_est);
+
+	// Acceleration update
+	filter_states->a_ned_est = filter_states_prev->a_ned_est + _filter_gains.H_v / (deltatime) *
+				   (v_measured - filter_states_prev->v_ned_est);
+	filter_states->saturate_acceleration(ACCELERATION_SATURATION);
+}
+
+void TargetEstimator::prediction_update(float deltatime)
+{
+	// Temporary copy to not mix old and new values during the update calculations
+	const matrix::Vector3f v_ned_est_prev = _filter_states.v_ned_est;
+	const matrix::Vector3f a_ned_est_prev = _filter_states.a_ned_est;
+
+	if (PX4_ISFINITE(v_ned_est_prev(0)) && PX4_ISFINITE(v_ned_est_prev(1)) && PX4_ISFINITE(v_ned_est_prev(2))) {
+		_filter_states.x_ned_est += deltatime * v_ned_est_prev + 0.5f * a_ned_est_prev * deltatime * deltatime;
+	}
+
+	if (PX4_ISFINITE(a_ned_est_prev(0)) && PX4_ISFINITE(a_ned_est_prev(1)) && PX4_ISFINITE(a_ned_est_prev(2))) {
+		_filter_states.v_ned_est += deltatime * a_ned_est_prev;
+	}
+}
+
+
+matrix::Vector3<double> TargetEstimator::get_lat_lon_alt_est() const
+{
+	matrix::Vector3<double> lat_lon_alt{NAN, NAN, NAN};
```

Comment:
```suggestion
	matrix::Vector3<double> lat_lon_alt{(double)NAN, (double)NAN, (double)NAN};
```

You'll need to cast these to make clang-tidy happy.